homeassistant/components/unifi_direct/device_tracker.py

```diff
@@ -37,104 +34,42 @@
+        self.update_clients()
+        return list(self.clients.keys())
 
-    def get_device_name(self, device):
+    def get_device_name(self, device: str) -> str | None:
         """Return the name of the given device or None if we don't know."""
-        hostname = next(
-            (
-                value.get("hostname")
-                for key, value in self.last_results.items()
-                if key.upper() == device.upper()
-            ),
-            None,
-        )
-        if hostname is not None:
-            hostname = str(hostname)
-        return hostname
-
-    def _connect(self):
-        """Connect to the Unifi AP SSH server."""
-
-        self.ssh = pxssh.pxssh(options={"HostKeyAlgorithms": "ssh-rsa"})
-        try:
-            self.ssh.login(
-                self.host, self.username, password=self.password, port=self.port
-            )
-            self.connected = True
-        except exceptions.EOF:
-            _LOGGER.error("Connection refused. SSH enabled?")
-            self._disconnect()
-
-    def _disconnect(self):
-        """Disconnect the current SSH connection."""
-        try:
-            self.ssh.logout()
-        except Exception:  # pylint: disable=broad-except
-            pass
-        finally:
-            self.ssh = None
-
-        self.connected = False
+        client_info = self.clients.get(device)
+        if client_info:
+            return client_info.get("hostname")
+        return None
 
-    def _get_update(self):
+    def update_clients(self) -> bool:
+        """Update the client info from AP."""
         try:
-            if not self.connected:
-                self._connect()
-            # If we still aren't connected at this point
-            # don't try to send anything to the AP.
-            if not self.connected:
-                return None
-            self.ssh.sendline(UNIFI_COMMAND)
-            self.ssh.prompt()
-            return self.ssh.before
-        except pxssh.ExceptionPxssh as err:
-            _LOGGER.error("Unexpected SSH error: %s", str(err))
-            self._disconnect()
-            return None
-        except (AssertionError, exceptions.EOF) as err:
-            _LOGGER.error("Connection to AP unavailable: %s", str(err))
-            self._disconnect()
-            return None
-
-
-def _response_to_json(response):
-    try:
-        json_response = json.loads(str(response)[31:-1].replace("\\", ""))
-        _LOGGER.debug(str(json_response))
-        ssid_table = json_response.get(UNIFI_SSID_TABLE)
-        active_clients = {}
-
-        for ssid in ssid_table:
-            client_table = ssid.get(UNIFI_CLIENT_TABLE)
-            for client in client_table:
-                active_clients[client.get("mac")] = client
-
-        return active_clients
-    except (ValueError, TypeError):
-        _LOGGER.error("Failed to decode response from AP")
-        return {}
+            self.clients = self.ap.get_clients()
+            return True
```

Comment:
Move the `return True` to the end of the method.

Reply:
done. thanks for the suggestion.